--- imgui-node-editor/imgui_canvas.cpp	2024-04-15 09:38:40.768877831 +0800
+++ imgui_src_build/imnode/src/imgui_canvas.cpp	2024-04-15 09:53:09.440727450 +0800
@@ -561,6 +561,21 @@
             m_DrawList->CmdBuffer.erase(m_DrawList->CmdBuffer.Data + m_DrawListCommadBufferSize);
         else if (m_DrawList->CmdBuffer.size() >= m_DrawListCommadBufferSize && m_DrawList->CmdBuffer[m_DrawListCommadBufferSize - 1].UserCallback == ImDrawCallback_ImCanvas)
             m_DrawList->CmdBuffer.erase(m_DrawList->CmdBuffer.Data + m_DrawListCommadBufferSize - 1);
+        // Proposed solution: test all commands from index >= m_DrawListFirstCommandIndex
+        // and remove the one with UserCallback == ImDrawCallback_ImCanvas
+        // (based on the original code, it seems there can be only one)
+        int idxCommand_ImDrawCallback_ImCanvas = -1;
+        for (int i = m_DrawListFirstCommandIndex; i < m_DrawList->CmdBuffer.size(); ++i)
+        {
+            auto & command = m_DrawList->CmdBuffer[i];
+            if (command.UserCallback == ImDrawCallback_ImCanvas)
+            {
+                idxCommand_ImDrawCallback_ImCanvas = i;
+                break;
+            }
+        }
+        if (idxCommand_ImDrawCallback_ImCanvas >= 0)
+            m_DrawList->CmdBuffer.erase(m_DrawList->CmdBuffer.Data + idxCommand_ImDrawCallback_ImCanvas);
     }
 
     auto& fringeScale = ImFringeScaleRef(m_DrawList);
