--- imgui-node-editor/imgui_extra_math.h	2024-04-06 19:17:36
+++ imgui_src_build/imnode/include/imgui_extra_math.h	2024-04-06 19:20:30
@@ -30,8 +30,10 @@
 
 
 //------------------------------------------------------------------------------
+# if IMGUI_VERSION_NUM < 19002
 inline bool operator==(const ImVec2& lhs, const ImVec2& rhs);
 inline bool operator!=(const ImVec2& lhs, const ImVec2& rhs);
+#endif
 inline ImVec2 operator*(const float lhs, const ImVec2& rhs);
 # if IMGUI_VERSION_NUM < 18955
 inline ImVec2 operator-(const ImVec2& lhs);
